grovedb/src/batch/mod.rs

```diff
@@ -909,12 +1330,150 @@ impl GroveDb {
+            ops,
+            batch_apply_options,
+            |_cost, _old_flags, _new_flags| Ok(false),
+            |_flags, removed_bytes| Ok(BasicStorageRemoval(removed_bytes)),
+            transaction,
+        )
+    }
+
+    pub fn open_batch_transactional_merk_at_path<'db, 'p, P>(
+        &'db self,
+        storage_batch: &'db StorageBatch,
+        path: P,
+        tx: &'db Transaction,
+        new_merk: bool,
+    ) -> CostResult<Merk<PrefixedRocksDbBatchTransactionContext<'db>>, Error>
+    where
+        P: IntoIterator<Item = &'p [u8]>,
+        <P as IntoIterator>::IntoIter: DoubleEndedIterator + Clone,
+    {
+        let mut path_iter = path.into_iter();
+        let mut cost = OperationCost::default();
+        let storage = self
+            .db
+            .get_batch_transactional_storage_context(path_iter.clone(), &storage_batch, tx)
+            .unwrap_add_cost(&mut cost);
+
+        match path_iter.next_back() {
+            Some(key) => {
+                if new_merk {
+                    Ok(Merk::open_empty(storage, MerkType::LayeredMerk)).wrap_with_cost(cost)
+                } else {
+                    let parent_storage = self
+                        .db
+                        .get_transactional_storage_context(path_iter.clone(), tx)
+                        .unwrap_add_cost(&mut cost);
+                    let element = cost_return_on_error!(
+                        &mut cost,
+                        Element::get_from_storage(&parent_storage, key).map_err(|_| {
+                            Error::InvalidPath("could not get key for parent of subtree".to_owned())
+                        })
+                    );
+                    if let Element::Tree(root_key, _) = element {
+                        Merk::open_layered_with_root_key(storage, root_key)
+                            .map_err(|_| {
+                                Error::CorruptedData(
+                                    "cannot open a subtree with given root key".to_owned(),
+                                )
+                            })
+                            .add_cost(cost)
+                    } else {
+                        Err(Error::CorruptedPath(
+                            "cannot open a subtree as parent exists but is not a tree",
+                        ))
+                        .wrap_with_cost(OperationCost::default())
+                    }
+                }
+            }
+            None => {
+                if new_merk {
+                    Ok(Merk::open_empty(storage, MerkType::BaseMerk)).wrap_with_cost(cost)
+                } else {
+                    Merk::open_base(storage)
+                        .map_err(|_| {
+                            Error::CorruptedData("cannot open a the root subtree".to_owned())
+                        })
+                        .add_cost(cost)
+                }
+            }
+        }
+    }
+
+    pub fn open_batch_merk_at_path<'a>(
```

Comment:
Since we are getting the root of the underlying Merk tree inside the parent Merk, dealing with accessing two data storages felt more comfortable in GroveDB.

Reply:
well inside this method it's not consistent either, batch and non-batch contexts are used

Reply:
Would it matter if we are only getting data from the parent? the batch just offsets writing afaik.

Reply:
Exactly what I'm trying to say: opening a merk and "batch" opening a merk has no real difference but we keep two of these implementations